protocols/floodsub/src/protocol.rs

```diff
@@ -28,14 +28,42 @@ use libp2p_core::{upgrade, InboundUpgrade, OutboundUpgrade, PeerId, UpgradeInfo}
 use prost::Message;
 use std::{io, iter, pin::Pin};
 
+const MAX_MESSAGE_LEN_BYTES: usize = 2048;
+
 /// Implementation of `ConnectionUpgrade` for the floodsub protocol.
-#[derive(Debug, Clone, Default)]
-pub struct FloodsubProtocol {}
+pub struct FloodsubProtocol {
+    codec: prost_codec::Codec<rpc_proto::Rpc>,
+}
 
 impl FloodsubProtocol {
     /// Builds a new `FloodsubProtocol`.
     pub fn new() -> FloodsubProtocol {
-        FloodsubProtocol {}
+        let codec = prost_codec::Codec::new(MAX_MESSAGE_LEN_BYTES);
+        FloodsubProtocol { codec }
+    }
+}
+
```

Comment:
These 3 impl's were added because the `#[derive(Debug, Clone, Default)]`. When that was there it complained the Codec didn't implement those. When I added these impls, then the need for the `Derive` went away.

I'm not sure this is the appropriate approach: It seems like Codec should handle this (e.g. it wasn't a problem in `libp2p-gossipsub`)